evap/results/views.py

```diff
@@ -143,12 +151,16 @@ def course_detail(request, semester_id, course_id):
             reviewer=request.user.is_reviewer,
             contributor=course.is_user_contributor_or_delegate(request.user),
             can_download_grades=request.user.can_download_grades,
-            public_view=public_view)
+            view=view,
+            contributor_id=contributor_id,
+        )
     return render(request, "results_course_detail.html", template_data)
 
 
-def user_can_see_text_answer(user, represented_users, text_answer, public_view=False):
-    if public_view:
+def user_can_see_text_answer(user, represented_users, text_answer, view='full'):
+    if view == 'public':
+        return False
+    if view == 'filter' and text_answer.is_private:
         return False
```

Comment:
shouldn't comments of other contributors `return False` as well?

Reply:
other contributors are completely removed, but for consistency: probably yes.

Reply:
if i understand correctly, you're referring to `{% if not view == 'filter' or view_as_user.id == contributor.id %}`. to me that specific check is the same case as below, technically it does not matter where we do the filtering, so i would do it at the earliest possible time, because why give data to the template where we know in the view already that it's not gonna be used?

regarding `user_can_see_text_answer`, i'd say a method is broken or not regardless of whether there is other code covering up the error ;) so yeah, i'm strongly in favor of fixing this.